boxsdk/session/box_session.py

```diff
@@ -81,7 +82,7 @@ def __init__(self, oauth, network_layer, default_headers=None):
         """
         self._oauth = oauth
         self._network_layer = network_layer
-        self._default_headers = default_headers or {}
+        self._default_headers = default_headers or {'User-Agent': Version.USER_AGENT_STRING}
```

Comment:
Could this instead be

```python
self._default_headers = dict(default_headers)
self._default_headers.setdefault('User-Agent', Version.USER_AGENT_STRING)
self._default_headers = {header: header_value for header, header_value in six.iteritems(self._default_headers) if header_value is not None}
```

That way, we include the `User-Agent` even if they've defined their own headers, but we still allow them to turn off the `User-Agent` by passing `None`.

Also, does `'User-Agent'` need to be a byte string?